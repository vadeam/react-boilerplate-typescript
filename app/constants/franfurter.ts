export const DEFAULT_BASE_CURRENCY = 'EUR'
