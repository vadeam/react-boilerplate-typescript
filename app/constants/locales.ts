export const LOCALE_DE = 'de'
export const LOCALE_EN = 'en'
export const DEFAULT_LOCALE = LOCALE_EN

export const appLocales: Locales[] = [LOCALE_EN, LOCALE_DE]
export const CUSTOM_LOCALE_STORAGE_KEY = 'custom-locale'
