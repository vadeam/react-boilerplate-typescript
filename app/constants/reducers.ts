export const REDUCER_KEY_APP = 'app'
export const REDUCER_KEY_LANGUAGE = 'language'
export const REDUCER_KEY_ROUTER = 'router'
export const REDUCER_KEY_HOME = 'home'
// for testing purposes
export const REDUCER_KEY_TEST = 'test'
