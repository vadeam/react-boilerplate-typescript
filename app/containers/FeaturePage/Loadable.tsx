/**
 * Asynchronously loads the component for FeaturePage
 */
import * as React from 'react'

import LoadingIndicator from 'components/LoadingIndicator'
import loadable from 'utils/loadable'

export default loadable(() => import('./index'), {
  fallback: <LoadingIndicator />,
})
