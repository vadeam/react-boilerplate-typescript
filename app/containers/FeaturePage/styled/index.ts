export * from './ListStyled'
export * from './ListItemStyled'
export * from './ListItemTitleStyled'
