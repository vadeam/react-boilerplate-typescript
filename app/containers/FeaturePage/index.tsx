/**
 * FeaturePage
 * List all the features
 */
import * as React from 'react'
import { Helmet } from 'react-helmet-async'
import { useIntl } from 'react-intl'

import H1 from 'components/H1'

import { ListStyled, ListItemStyled, ListItemTitleStyled } from './styled'
import messages from './messages'

const FeaturePage = () => {
  const { formatMessage } = useIntl()
  return (
    <div>
      <Helmet>
        <title>{formatMessage({ ...messages.title })}</title>
        <meta name="description" content="Feature page of React Boilerplate TypeScript application" />
      </Helmet>
      <H1>{formatMessage({ ...messages.header })}</H1>
      <ListStyled>
        <ListItemStyled>
          <ListItemTitleStyled>{formatMessage({ ...messages.scaffoldingHeader })}</ListItemTitleStyled>
          <p> {formatMessage({ ...messages.scaffoldingMessage })}</p>
        </ListItemStyled>

        <ListItemStyled>
          <ListItemTitleStyled>{formatMessage({ ...messages.feedbackHeader })}</ListItemTitleStyled>
          <p>{formatMessage({ ...messages.feedbackMessage })}</p>
        </ListItemStyled>
        <ListItemStyled>
          <ListItemTitleStyled>{formatMessage({ ...messages.feedbackHeader })}</ListItemTitleStyled>
          <p>{formatMessage({ ...messages.feedbackMessage })}</p>
        </ListItemStyled>

        <ListItemStyled>
          <ListItemTitleStyled>{formatMessage({ ...messages.routingHeader })}</ListItemTitleStyled>
          <p>{formatMessage({ ...messages.routingMessage })}</p>
        </ListItemStyled>

        <ListItemStyled>
          <ListItemTitleStyled>{formatMessage({ ...messages.networkHeader })}</ListItemTitleStyled>
          <p>{formatMessage({ ...messages.networkMessage })}</p>
        </ListItemStyled>

        <ListItemStyled>
          <ListItemTitleStyled>{formatMessage({ ...messages.intlHeader })}</ListItemTitleStyled>
          <p>{formatMessage({ ...messages.intlMessage })}</p>
        </ListItemStyled>
      </ListStyled>
    </div>
  )
}

export default FeaturePage
