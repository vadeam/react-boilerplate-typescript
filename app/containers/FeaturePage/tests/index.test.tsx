import { render } from '@testing-library/react'
import React from 'react'
import { HelmetProvider } from 'react-helmet-async'
import { IntlProvider } from 'react-intl'

import { LOCALE_EN } from 'constants/locales'

import FeaturePage from '../index'

describe('<FeaturePage />', () => {
  it('should render its heading', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={LOCALE_EN}>
        <HelmetProvider>
          <FeaturePage />
        </HelmetProvider>
      </IntlProvider>,
    )

    expect(firstChild).toMatchSnapshot()
  })
})
