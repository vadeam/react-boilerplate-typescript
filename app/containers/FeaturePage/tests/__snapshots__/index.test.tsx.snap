// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturePage /> should render its heading 1`] = `
<div>
  <h1
    class="sc-bdfBQB eyZqMq"
  >
    Features
  </h1>
  <ul
    class="sc-gsTEea kJAHDg"
  >
    <li
      class="sc-dlfnuX cQWiti"
    >
      <p
        class="sc-hKgJUU jQXVEd"
      >
        Quick scaffolding
      </p>
      <p>
         
        Automate the creation of components, containers, routes, selectors
  and sagas - and their tests - right from the CLI!
      </p>
    </li>
    <li
      class="sc-dlfnuX cQWiti"
    >
      <p
        class="sc-hKgJUU jQXVEd"
      >
        Instant feedback
      </p>
      <p>
        
      Enjoy the best DX and code your app at the speed of thought! Your
    saved changes to the CSS and JS are reflected instantaneously
    without refreshing the page. Preserve application state even when
    you update something in the underlying code!
    
      </p>
    </li>
    <li
      class="sc-dlfnuX cQWiti"
    >
      <p
        class="sc-hKgJUU jQXVEd"
      >
        Instant feedback
      </p>
      <p>
        
      Enjoy the best DX and code your app at the speed of thought! Your
    saved changes to the CSS and JS are reflected instantaneously
    without refreshing the page. Preserve application state even when
    you update something in the underlying code!
    
      </p>
    </li>
    <li
      class="sc-dlfnuX cQWiti"
    >
      <p
        class="sc-hKgJUU jQXVEd"
      >
        Industry-standard routing
      </p>
      <p>
        
      Write composable CSS that's co-located with your components for
    complete modularity. Unique generated class names keep the
    specificity low while eliminating style clashes. Ship only the
    styles that are on the page for the best performance.
    
      </p>
    </li>
    <li
      class="sc-dlfnuX cQWiti"
    >
      <p
        class="sc-hKgJUU jQXVEd"
      >
        Offline-first
      </p>
      <p>
        
      The next frontier in performant web apps: availability without a
      network connection from the instant your users load the app.
    
      </p>
    </li>
    <li
      class="sc-dlfnuX cQWiti"
    >
      <p
        class="sc-hKgJUU jQXVEd"
      >
        Complete i18n Standard Internationalization & Pluralization
      </p>
      <p>
        Scalable apps need to support multiple languages, easily add and support multiple languages with \`react-intl\`.
      </p>
    </li>
  </ul>
</div>
`;
